test/images/webhook/main.go

```diff
@@ -115,12 +157,20 @@ func serve(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func servePods(w http.ResponseWriter, r *http.Request) {
+	serve(w, r, "pods")
```

Comment:
Maybe pass the function rather than a resource name. You would then get to lose the switch above.

Reply:
Done.